### PR TITLE
Aircrafts and Sites

### PR DESCRIPTION
**Pull request by german-e-mas**
In this PR I include some of the classes of the Model layer. They represent Aircrafts and Landing Sites. 

Specific types of aircraft, such as large planes, light planes and helicopters are extension to a base `Aircraft` class. Same happens with specific landing sites, that extend the `LandingSite` class.

The `land` method on the Aircraft returns a boolean that reflects whether it can land on the corresponding site given. The method calls the site's `accept` method, passing the aircraft as an argument. Each landing site type has overloaded methods for each type of aircraft, and returns whether it can land there or not.

Having this, we can see that, for example:

```
Aircraft lightPlane = new LightPlane();
Aircraft helicopter = new Helicopter();
LandingSite longRunway = new LongRunway();
LandingSite helipad = new Helipad();

lightPlane.land(longRunway); // Returns true
lightPlane.land(helipad); // Returns false
helicopter.land(longRunway); // Returns false
helicopter.land(helipad); // Returns true
```

ptal @pablisho

### CODE REVIEW

File: app/src/main/java/mas/german/landingplanes/Aircraft.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+/**
+ * Represents an Aircraft. Specific aircraft types should extend from this class.
+ * An Aircraft has speed, direction and size. An Aircraft can land on Landing Sites.
+ */
+public abstract class Aircraft {
+    private static final String TAG = Aircraft.class.getSimpleName();
+
+    private int mSpeed;
+    private double mDirection;
+    private int mRadius;
```
Comment (pablisho) [inline app/src/main/java/mas/german/landingplanes/Aircraft.java]: No position?

File: app/src/main/java/mas/german/landingplanes/LandingSite.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+/**
+ * Represents a Landing Site. Specific landing sites should extend from this class.
+ * A landing site has a size (width and height) and can accept Aircrafts for landing on them.
+ */
+public abstract class LandingSite {
+    private static final String TAG = LandingSite.class.getSimpleName();
+
+    private int mWidth;
+    private int mHeight;
```
Comment (pablisho) [inline app/src/main/java/mas/german/landingplanes/LandingSite.java]: No position?

File: app/src/main/java/mas/german/landingplanes/LongRunway.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+/**
+ * Represents a Long Runway, which is a Landing Site.
+ * Large and light planes can land on it.
+ */
+public class LongRunway extends LandingSite {
+    private static final String TAG = LongRunway.class.getSimpleName();
+
+    public boolean accept(LargePlane largePlane) {
+        // Planes can land on Large Runways.
+        return true;
```
Comment (pablisho) [inline app/src/main/java/mas/german/landingplanes/LongRunway.java]: The double dispatch is fine! But I would like to see the landing logic here too. It should return true only if the plane is close enough to the landing site.

**Comment by german-e-mas** [comment]
Thanks for the review @pablisho ! In this PR I didn't include position of the aicrafts/landing sites. I wanted to see if I was going through the right path before adding things. I was experimenting with lands, crashes and position changes using a Game class. My following PR will cover that.

File: app/src/main/java/mas/german/landingplanes/LongRunway.java (hunk 3 above)
Comment (german-e-mas) [inline app/src/main/java/mas/german/landingplanes/LongRunway.java]: I was thinking about checking the positions outside this method. For example, if a plane is close enough to a site, then call `land`.

In my periodic task, for each `Aircraft` I have, I compare their position to all `LandingSites`. If there is one close enough, I call the `land` method of that aircraft (which can be true or false, depending on the type). Do you think I should call the `land` method for every site, and have it check the position internally?

Comment (pablisho) [inline app/src/main/java/mas/german/landingplanes/LongRunway.java]: Yes, whenever you find yourself asking for a field value and processing it outside it's a signal you are doing something wrong. In pure OOP you shall never ask for the values, you have methods that process them. Also consider extra classes for the Position and that class could have methods to say if two positions are close enough to land or collide.

**Pull request merged**